### PR TITLE
D2: `subject` field now uses its own LiveData

### PR DESCRIPTION
**Pull request by KevinBoulongne**
Depends on #1783

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/AiPropositionFragment.kt [hunk 1]
```diff
@@ -165,15 +168,18 @@ class AiPropositionFragment : Fragment() {
 
     private fun choosePropositionAndPopBack() = with(aiViewModel) {
 
-        fun applyProposition(subject: String?, content: String) {
+        fun applyProposition(subject: String?, content: String) = with(newMessageViewModel) {
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/AiPropositionFragment.kt:171]: You use a `with(newMessageViewModel)`only for the liveData.

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt [hunk 2]
```diff
@@ -552,6 +552,10 @@ class NewMessageFragment : Fragment() {
         }
     }
 
+    fun formatSubject(): String? {
+        return binding.subjectTextField.text?.toString()?.ifBlank { null }
+    }
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt:557]:
```suggestion
    fun formatSubject() = binding.subjectTextField.text?.toString()?.ifBlank { null }
```

**Review by tevincent — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/AiPropositionFragment.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/newMessage/AiPropositionFragment.kt:171]: Will be used for a 2nd one in the next PR.

**Review by LunarX — CHANGES_REQUESTED** [review]

**Review by tevincent — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt [hunk 3]
```diff
@@ -98,6 +98,11 @@ class NewMessageViewModel @Inject constructor(
     private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
 
     var draftInRAM: Draft = Draft()
+
+    //region UI data
+    val subjectLiveData = SingleLiveEvent<String?>()
```
[the remark this answers is not in the document]
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt:103]: I agree on the naming issue with @FabianDevel. It used to be called something along the lines of "subjectToInsert" or something if I'm not mistaken, either this or something else mentioning insertion or AI maybe

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt [hunk 4]
```diff
@@ -535,7 +532,7 @@ class NewMessageFragment : Fragment() {
             requireActivity().finishAppAndRemoveTaskIfNeeded()
         }
 
-        if (newMessageViewModel.draftInRAM.subject.isNullOrBlank()) {
+        if (binding.subjectTextField.text?.isBlank() == true) {
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt:535]: Which call in the AiManager ?

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt]: 👌 
This has been reverted to use the previous name.

**Review by FabianDevel — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt [hunk 5]
```diff
@@ -98,6 +98,9 @@ class NewMessageViewModel @Inject constructor(
     private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
 
     var draftInRAM: Draft = Draft()
+
+    val aiOutputToInsert = SingleLiveEvent<Pair<String?, String>>()
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt:102]: Why did you need to move `aiOutputToInsert` from AiViewModel to NewMessageViewModel? Can it not stay there?

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt [hunk 6]
```diff
@@ -475,12 +466,20 @@ class NewMessageFragment : Fragment() {
         }
     }
 
+    private fun observeAiOutput() = with(binding) {
+        newMessageViewModel.aiOutputToInsert.observe(viewLifecycleOwner) { (subject, content) ->
+            subject?.let(subjectTextField::setText)
+            bodyText.setText(content)
+        }
+    }
+
```
Comment (LunarX) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt:475]: Can't the observe stay in the NewMessageAiManager class?

**Review by LunarX — CHANGES_REQUESTED** [review]
